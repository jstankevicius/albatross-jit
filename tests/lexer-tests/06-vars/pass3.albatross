void function(char argument) {
  string ret;
  if ( argument == 10)
    return 10;
  else
    return function(argument-1);
}

void anotherFunction() {
  int[] argument;
  while (1) { return; }
}

int oneMoreFunction() {
  while (0) { return 01; } else { return 10; }
}

int result := 0;

if (1) { result := 1; }

if (0) { result := 10; } else { result := 1000; }
