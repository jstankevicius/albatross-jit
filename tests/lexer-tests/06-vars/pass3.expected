1 1 TYPE void
6 1 NAME function TYPE void
14 1 (
15 1 TYPE char
20 1 NAME argument TYPE char
28 1 )
30 1 {
3 2 TYPE string
10 2 NAME ret TYPE string
13 2 SEMICOLON
3 3 IF
6 3 (
8 3 NAME argument TYPE string
17 3 EQ
20 3 INT 10
22 3 )
5 4 RETURN
12 4 INT 10
14 4 SEMICOLON
3 5 ELSE
5 6 RETURN
12 6 NAME function TYPE string
20 6 (
21 6 NAME argument TYPE string
29 6 MINUS
30 6 INT 1
31 6 )
32 6 SEMICOLON
1 7 }
1 9 TYPE void
6 9 NAME anotherFunction TYPE void
21 9 (
22 9 )
24 9 {
3 10 TYPE int
6 10 [
7 10 ]
9 10 NAME argument TYPE int
17 10 SEMICOLON
3 11 WHILE
9 11 (
10 11 INT 1
11 11 )
13 11 {
15 11 RETURN
21 11 SEMICOLON
23 11 }
1 12 }
1 14 TYPE int
5 14 NAME oneMoreFunction TYPE int
20 14 (
21 14 )
23 14 {
3 15 WHILE
9 15 (
10 15 INT 0
11 15 )
13 15 {
15 15 RETURN
22 15 INT 1
24 15 SEMICOLON
26 15 }
28 15 ELSE
33 15 {
35 15 RETURN
42 15 INT 10
44 15 SEMICOLON
46 15 }
1 16 }
1 18 TYPE int
5 18 NAME result TYPE int
12 18 ASSIGN
15 18 INT 0
16 18 SEMICOLON
1 20 IF
4 20 (
5 20 INT 1
6 20 )
8 20 {
10 20 NAME result TYPE int
17 20 ASSIGN
20 20 INT 1
21 20 SEMICOLON
23 20 }
1 22 IF
4 22 (
5 22 INT 0
6 22 )
8 22 {
10 22 NAME result TYPE int
17 22 ASSIGN
20 22 INT 10
22 22 SEMICOLON
24 22 }
26 22 ELSE
31 22 {
33 22 NAME result TYPE int
40 22 ASSIGN
43 22 INT 1000
47 22 SEMICOLON
49 22 }
